src/runtime/HalideBuffer.h

```diff
@@ -467,10 +473,18 @@ class Buffer {
         }
     }
 
-    /** Make a Buffer<T> from another Buffer<T> of possibly-different
-     * dimensionality and type. Asserts if D is less than the
-     * dimensionality of the argument, or if there's a type
-     * mismatch. */
+    /** Copy constructor. Does not copy underlying data. */
+    Buffer(const Buffer<T, D> &other) : buf(other.buf),
```

Comment:
Do we want this to be explicit? If not, should we add a comment saying it's deliberate? (IMHO it's probably better to tag all single-arg ctors as either 'explicit' or /*not-explicit*/ to indicate that it's been considered, but maybe I'm overthinking it...)